Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -796,6 +813,47 @@ open class ComposerVC: _ViewController,
             log.assertionFailure(error.localizedDescription)
         }
     }
+    
+    open func checkChannelCountdown() {
+        if content.countdownDuration > 0 {
+            var duration = content.countdownDuration
+            let timer = Timer.scheduledTimer(
+                withTimeInterval: 1,
+                repeats: true
+            ) { [weak self] timer in
+                guard let self = self else { return }
+                if duration == 0 {
+                    timer.invalidate()
+                        
+                    self.composerView.countdownButton.isHidden = true
+                        
+                    if self.content.state == .edit {
+                        self.composerView.confirmButton.isHidden = false
+                    } else {
+                        self.composerView.sendButton.isHidden = false
+                    }
+                        
+                    self.composerView.inputMessageView.textView.placeholderLabel.text = L10n.Composer.Placeholder.message
+                    self.composerView.isUserInteractionEnabled = true
+                } else {
+                    self.composerView.sendButton.isHidden = true
+                    self.composerView.countdownButton.isHidden = false
+                    self.composerView.confirmButton.isHidden = true
+                        
+                    self.composerView.inputMessageView.textView.placeholderLabel.text = L10n.Composer.Placeholder.slowMode
+                    self.composerView.isUserInteractionEnabled = false
+                        
+                    self.composerView.countdownButtonWidthConstraint.constant = duration >= 10 ? 40 : 32
```

Comment:
We should probably extract the 40 and 32 values from magic numbers to properties that can be customized in ComposerVC 👍

Reply:
Removed. Not needed now 😃